eng/targetingpacks.targets

```
@@ -147,6 +147,17 @@
     </ItemGroup>
   </Target>
 
+  <!-- Use local Crossgen2 pack for NetCoreAppCurrent. -->
+  <Target Name="UpdateLocalCrossgen2Pack"
+          Condition="'$(UseLocalCrossgen2Pack)' == 'true'"
+          AfterTargets="ResolveFrameworkReferences">
+    <ItemGroup>
+      <ResolvedCrossgen2Pack PackageDirectory="$(CoreCLRArtifactsPath)\$(BuildArchitecture)\crossgen2\"
```

Let's not hardcode this path here and use a property instead. I noticed people copying this file to use a local build tree which wouldn't work anymore then.

```suggestion
      <ResolvedCrossgen2Pack PackageDirectory="$(Crossgen2Dir)"
```

---

Btw, what made me not pursue this further is that the local build layout doesn't group the binaries under a "tools" folder which the `ResolveReadyToRunCompilers` task depends on?

---

This PR also updates `crossgen2_inbuild.targets` to put it under a `tools` folder.  I need to update one more place I realized I missed related to this.